### PR TITLE
Fixed triple-click only selecting link when paragraph starts with a link

### PR DESCRIPTION
**Pull request by kevinansfield**
closes https://github.com/facebook/lexical/issues/5045

- modified the triple-click selection adjustment that removes unwanted trailing element to be aware of links as they create a deeper child/parent relationship than the previous code expected
- for now this commit uses a workaround of modifying the DOM selection directly because attempts to detect the paragraph element and use `paragraphNode.select(0)` resulted in an odd state where visually the selection looked correct and text-level formatting could be applied as expected but block-level changes like changing to a header had no effect

### CODE REVIEW

File: packages/lexical/src/LexicalEvents.ts [hunk 1]
```diff
@@ -393,17 +393,35 @@ function onClick(event: PointerEvent, editor: LexicalEditor): void {
...
+              normalizedAnchorElement &&
+              normalizedAnchorElement !== normalizedFocusElement
+            ) {
+              // use DOM range because paragraphNode.select(0) results in a selection that
+              // visually appears to cover the link+text but can't have block-level formatting applied.
+              // FIXME: needs a deeper fix?
+              const range = document.createRange();
+              range.selectNodeContents(normalizedAnchorElement);
+              domSelection.removeAllRanges();
+              domSelection.addRange(range);
```
Comment (fantactuka) [inline packages/lexical/src/LexicalEvents.ts:424]: I wonder if we could instead look up for nearest non-inline element:

```
const selectableNode = $findMatchingParent(
  anchorNode,
  (node) => $isElementNode(node) && !node.isInline(),
);

if (selectableNode !== null) {
  selectableNode.select(0);
}
```

Comment (kevinansfield) [reply packages/lexical/src/LexicalEvents.ts:424]: For comparison...

State when creating selection via DOM:
```
 root
  ├ (11) paragraph  
> | ├ (8) link  "https://" { rel: noreferrer }
> | | └ (9) text  "One"
> | |              ^^^
> | └ (10) text  " two"
> |               ^^^^
  └ (4) paragraph  
    └ (5) text  "three"

 selection: range  
  ├ anchor { key: 9, offset: 0, type: text }
  └ focus { key: 10, offset: 4, type: text }
```

State when creating selection via `paragraphNode.select(0)`:

```
 root
  ├ (2) paragraph  
> | ├ (7) link  "https://" { rel: noreferrer }
> | | └ (3) text  "One"
> | └ (6) text  " two"
  └ (4) paragraph  
    └ (5) text  "three"

 selection: range  
  ├ anchor { key: 2, offset: 0, type: element }
  └ focus { key: 6, offset: 4, type: text }
```

Key difference being the anchor where it's the paragraph node with `.select(0)` but the beginning of the link's text node with dom selection. The latter works with block modifications where the former doesn't.

Comment (kevinansfield) [reply packages/lexical/src/LexicalEvents.ts:424]: What's the expectation here regarding selection - should `.select(0)` create something that resembles the DOM-derived selection, or should both of those selections be effectively the same in terms of further manipulation?

Comment (ivailop7) [reply packages/lexical/src/LexicalEvents.ts:424]: @fantactuka can you have second look here, when you have a chance

**Comment by ivailop7** [comment]
This issue is still outstanding, would still be nice to fix.

**Pull request closed without merging**